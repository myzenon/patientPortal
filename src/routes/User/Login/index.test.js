import React from 'react'
import { Login } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const wrapper = render (
        <Login />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks on "Sign In" button | call login api', (done) => {
    const mockEmail = 'test@test.com'
    const mockPassword = 'A12345678*a'
    const props = {
        login: (email, password) => {
            expect(email).toBe(mockEmail)
            expect(password).toBe(mockPassword)
            done()
            return Promise.resolve()
        },
        page: {
            error: {
                status: null
            }
        }
    }
    const wrapper = mount (
        <Login {...props} />
    )
    wrapper.find('input[name="email"]').simulate('change', {
        target: {
            name: 'email',
            value: mockEmail
        }
    })
    wrapper.find('input[name="password"]').simulate('change', {
        target: {
            name: 'password',
            value: mockPassword
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
})

test('login failed | email or password incorrect', (done) => {
    const props = {
        login: () => Promise.resolve(),
        page: {
            error: {
                status: 401
            }
        }
    }
    const wrapper = mount (
        <Login {...props} />
    )
    wrapper.find('input[name="email"]').simulate('change', {
        target: {
            name: 'email',
            value: 'test@test.com'
        }
    })
    wrapper.find('input[name="password"]').simulate('change', {
        target: {
            name: 'password',
            value: 'A12345678*a'
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
    setTimeout(() => {
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0) 
})