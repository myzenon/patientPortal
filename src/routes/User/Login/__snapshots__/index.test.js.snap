// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login failed | email or password incorrect 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <p>
      Member Sign In
    </p>
    <p>
      <span
        class="newRegistration"
      >
        Don't have an account?  
        <a
          href="/welcome-new"
        >
          <u>
            Register now.
          </u>
        </a>
      </span>
    </p>
    <div
      class="formWrapper"
    >
      <form
        class=""
      >
        <span>
          <input
            class="inputText"
            name="email"
            placeholder="email"
            type="email"
            value="test@test.com"
          />
          <div
            class="textError"
          >
            Error: Wrong email or password.
          </div>
        </span>
        <div
          class="passwordwrapper"
        >
          <input
            autocomplete="a-password"
            class="passwordText"
            ispassword="true"
            name="password"
            placeholder="password"
            type="text"
            value="A12345678*a"
          />
          <button
            class="showHideButton btn btn-secondary"
            id="btnShowPassword"
            type="button"
          >
            Hide 
          </button>
        </div>
        <div>
          <button
            class="button btn btn-secondary"
            id="btnlogin"
            type="submit"
          >
            Sign In
          </button>
        </div>
      </form>
      <div>
        <span
          class="forgotPasswordText"
        >
          I forgot my password
        </span>
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <p>
      Member Sign In
    </p>
    <p>
      <span
        class="newRegistration"
      >
        Don't have an account?  
        <a
          href="/welcome-new"
        >
          <u>
            Register now.
          </u>
        </a>
      </span>
    </p>
    <div
      class="formWrapper"
    >
      <form
        class=""
      >
        <span>
          <input
            autofocus=""
            class="inputText"
            name="email"
            placeholder="email"
            type="email"
            value=""
          />
        </span>
        <div
          class="passwordwrapper"
        >
          <input
            autocomplete="a-password"
            class="passwordText"
            ispassword="true"
            name="password"
            placeholder="password"
            type="password"
            value=""
          />
          <button
            class="showHideButton btn btn-secondary"
            id="btnShowPassword"
            type="button"
          >
            Show
          </button>
        </div>
        <div>
          <button
            class="button btn btn-secondary"
            id="btnlogin"
            type="submit"
          >
            Sign In
          </button>
        </div>
      </form>
      <div>
        <span
          class="forgotPasswordText"
        >
          I forgot my password
        </span>
      </div>
      <div />
    </div>
  </div>
</div>
`;
